canary/consumer-java/src/main/java/com/amazon/kinesis/video/canary/consumer/WebrtcStorageCanaryConsumer.java

```diff
@@ -0,0 +1,227 @@
+import java.text.MessageFormat;
+import com.amazonaws.services.cloudwatch.model.Dimension;
+import com.amazonaws.services.cloudwatch.model.MetricDatum;
+import com.amazonaws.services.cloudwatch.model.PutMetricDataRequest;
+import com.amazonaws.services.cloudwatch.model.StandardUnit;
+
+
+
+
+
+import lombok.extern.log4j.Log4j2;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Optional;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.ArrayList;
+
+
+@Slf4j
+public class WebrtcStorageCanaryConsumer {
+
+    private Integer fragmentListLength = 0;
+
+    private static void getIntervalMetrics(CanaryFragmentList fragmentList, Date canaryStartTime, String streamName, SystemPropertiesCredentialsProvider credentialsProvider, String dataEndpoint, String region){
+        System.out.println("12 sec have passed...");
+        try{
+            TimestampRange timestampRange = new TimestampRange();
+            timestampRange.setStartTimestamp(canaryStartTime);
+            timestampRange.setEndTimestamp(new Date());
+
+            FragmentSelector fragmentSelector = new FragmentSelector();
+                fragmentSelector.setFragmentSelectorType("SERVER_TIMESTAMP");
+                fragmentSelector.setTimestampRange(timestampRange);
+
+            Boolean newFragmentReceived = false;
+
+            ExecutorService executorService = Executors.newFixedThreadPool(10);
+            Future<List<CanaryFragment>> listFragmentResult = executorService.submit(new CanaryListFragmentWorker(streamName, credentialsProvider, dataEndpoint, Regions.fromName(region), fragmentSelector));
+            List<CanaryFragment> newFragmentList = listFragmentResult.get();
+
+            if (newFragmentList.size() > fragmentList.getFragmentList().size())
+            {
+                newFragmentReceived = true;
+            }
+            System.out.println(MessageFormat.format("newFragmentReceived: {0}", newFragmentReceived));
+
+            fragmentList.setFragmentList(newFragmentList);
+
+            final AmazonCloudWatchAsync cwClient = AmazonCloudWatchAsyncClientBuilder.standard()
+                .withRegion(region)
+                .withCredentials(credentialsProvider)
+                .build();
+
+            final Dimension dimensionPerStream = new Dimension()
+                .withName("ProducerSDKCanaryStreamName")
+                .withValue(streamName);
+            List<MetricDatum> datumList = new ArrayList<>();
+
+            MetricDatum datum = new MetricDatum()
+                .withMetricName("FragmentReceived")
+                .withUnit(StandardUnit.None)
+                .withValue(newFragmentReceived ? 1.0 : 0.0)
+                .withDimensions(dimensionPerStream);
+            datumList.add(datum);
+
+            PutMetricDataRequest request = new PutMetricDataRequest()
+                .withNamespace("KinesisVideoSDKCanary")
```

Comment:
Should the WebRTC Storage canary have a different name?

Reply:
Will use the same CW namespace but will use a different dimension